src/main/java/hillel/spring/doctor/domain/DoctorRecord.java

```diff
@@ -0,0 +1,41 @@
+package hillel.spring.doctor.domain;
+
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.Id;
+import java.time.LocalDateTime;
+
+/* This class stores data
+   about all appointments of doctors
+   There are no Doctor and Pet entity fields,
+   just IDs. The reason is to prevent from EAGER loading
+   of huge DoctorRecord entity lists.
+   Cascade deleting are not supported, all record history must remain,
+   even if related Doctor or Pet entity has been deleted
+ */
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+@Builder(toBuilder = true)
+@Entity
+public class DoctorRecord {
+    @Id
+    @GeneratedValue
+    private Integer id;
+    private Integer doctorId;
+    private Integer petId;
+    private LocalDateTime startDate;
+
+    public Integer getStartHour() {
+        if (startDate == null) {
+            return null;
```

Comment:
I would say it is an invalid record if the start date is null.
And if the return type is nullable, it should be wrapped by Optional.
If we assume that start date is nullable, this could as following:
```
Optional<Integer> getStartHour() {
    return Optional.ofNullable(startDate).map(LocalDateTime::getHour)
}
```